server/api/src/rsFileChksum.cpp

```diff
@@ -339,11 +349,29 @@ int file_checksum(RsComm* _comm,
         }
     }};
 
-    char buffer[SVR_MD5_BUF_SZ];
+    // get the buffer size from server configuration - default to 1 MiB
+    int read_buffer_size;
+    try {
+        read_buffer_size = irods::get_advanced_setting<int>("checksum_read_buffer_size_in_bytes");
+    }
+    catch (const irods::exception&) {
+        log_api::debug("{} :: checksum_read_buffer_size_in_bytes not set. Using default of 1 MiB.", __func__);
+        read_buffer_size = 1024 * 1024;
+    }
+
+    if (read_buffer_size < 0 || read_buffer_size > INT_MAX) {
```

Comment:
Is there a reason this logic isn't in `fileChksum`?

Reply:
I just missed it / forgot about it.

It's done.